src/Transport/Config/AzureServiceBusTransport.cs

```diff
@@ -20,7 +21,25 @@ protected override TransportReceivingConfigurationResult ConfigureForReceiving(T
             return new TransportReceivingConfigurationResult(
                 Topology.GetMessagePumpFactory(),
                 Topology.GetQueueCreatorFactory(),
-                () => Task.FromResult(StartupCheckResult.Success) //TODO: figure out what this is for
+                () =>
+                {
+                    // Useless for me: when we resolve INamespacePartitioningStrategy to apply startup check, we have already applied checks defined into ctor of concrete classes. Right?
+                    var namespacePartitioningStrategy = Container.Resolve<INamespacePartitioningStrategy>();
+                    var namespaces = context.Settings.Get<List<string>>(WellKnownConfigurationKeys.Topology.Addressing.Partitioning.Namespaces);
+                    var endpointName = context.Settings.Get<EndpointName>();
+
+                    var configuredNamespaces = namespacePartitioningStrategy.GetNamespaces(endpointName.ToString(), PartitioningIntent.Creating);
+
+                    if (namespaces.Count != configuredNamespaces.Count())
+                        return Task.FromResult(StartupCheckResult.Failed("..."));
+
+                    var namespaceManager  = Container.Resolve<IManageNamespaceManagerLifeCycle>()
+                        .Get(context.ConnectionString);
```

Comment:
Might be wrong here, but wouldn't we need to run this on all the registered `namespaces`?